src/lib/storage/dictionary_compression.hpp

```diff
@@ -24,7 +26,8 @@ class DictionaryCompression {
    * @param column needs to be of type ValueColumn<T>
    * @return a compressed column of type DictionaryColumn<T>
    */
-  static std::shared_ptr<BaseColumn> compress_column(DataType data_type, const std::shared_ptr<BaseColumn>& column);
+  static std::tuple<std::shared_ptr<BaseColumn>, std::shared_ptr<opossum::BaseChunkColumnStatistics>> compress_column(
```

Comment:
Please move these additions to `ChunkEncoder` as `DictionaryCompression` is going to be removed very soon :)